tests/5.1/dispatch/test_dispatch_is_device_ptr.c

```diff
@@ -0,0 +1,74 @@
+//===---- test_dispatch_is_device_ptr.c -----------------------------------------===//
+// 
+// OpenMP API Version 5.1
+//
+// Uses dispatch construct as context for variant directive. Uses is_device_ptr
+// clause to ensure that the list item is in the device region.
+//
+// Inspired by "OpenMP 5.1 Features: The Dispatch Construct" video:
+// https://www.youtube.com/watch?v=ruugaX95gIs
+// 
+//===-------------------------------------------------------------------------------===//
+
+#include <omp.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <math.h>
+#include "ompvv.h"
+#include <stdbool.h>
+
+#define N 1024
+
+int errors;
+int i = 0;
+int *arr;
+
+void add_dev(int *arr);
+
+#pragma omp declare variant(add_dev) match(construct={dispatch}) 
+void add(int *arr){
+    #pragma omp parallel for
```

Comment:
If the implementation is permitted to call the base function inside the 'dispatch' construct – then the following must  be executed in a **target** region. _See comment further below._
```suggestion
    #pragma omp target parallel for is_device(arr)
```